File: tests/go/cases/case_266_thread_pool_capture_06.go
package escape_tests

var retainedCase266 = []map[string]string{}

func Case266ThreadPoolCapture06(input string) string {
	raw := input
	if raw == "" {
		raw = "sample"
	}
	payload := map[string]string{
		"task": "thread_pool_capture_06",
		"entity": "extreme",
		"stage": "stress",
		"input": raw,
		"checksum": "thread_pool_capture_06:" + raw,
	}
	// ESCAPE: goroutine captures payload and retains it.
	go func(v map[string]string) { retainedCase266 = append(retainedCase266, v) }(payload)
	return "ok"
}

File: tests/nodejs/cases/case_266_thread_pool_capture_06.js
/**
 * Task case 266: thread_pool_capture_06 deep stress pattern.
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case266ThreadPoolCapture06(input) {
  const taskName = 'thread_pool_capture_06';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'extreme',
    stage: 'stress',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  // ESCAPE: async task captures payload.
  setTimeout(() => retainedAudit.push(payload), 0);
  return 'ok';
}

module.exports = { case266ThreadPoolCapture06 };
